improve sponsor update process with schema definition and image path autocomplete

KyleMit opened the pull request:
Couple Dev Experience tweaks to updating the `sponsor.json` file

Saw @zekefarwell's note that that we might work through some more responsive CSS for logos.  In the meantime, I added a schema definition to `.vscode/settings.json` which should help ensure we're using one of the appropriate classes:

![image](https://user-images.githubusercontent.com/4307307/60745271-8e46a680-9f47-11e9-9083-76e04b34b09a.png)

I was trying to get [validation and autocomplete](https://stackoverflow.com/q/56907655/1366033) wired up on the logo path value.  There doesn't seem to be an easy way to validate that field explicitly, but I added a [recommended extension](https://marketplace.visualstudio.com/items?itemName=ionutvmi.path-autocomplete) in the `.vscode/extensions/json` file which provides autocomplete on relative paths:

![image](https://user-images.githubusercontent.com/4307307/60745357-e382b800-9f47-11e9-82b4-5e49a27c4444.png)

If you think these'll help our authorship of sponsor logos, we can merge in

zekefarwell commented:
The json schema making the `shape` field and enum is great!  Didn't know about that.

For the logo path I think I prefer the simplicity of it just being the filename even though the autocomplete is cool.  Kind a weird to have to slice off the first three characters of the path since it has to be a relative path.  But if you find the benefit of autocomplete worth the trade off, it doesn't matter too much to me.

I definitely do want a generalized set of CSS rules for the logos.  What I wrote so far is crap, but didn't want to wait to come up with the perfect solution for the logos to be displayed on the site.  I apologize for deploying code that I'm sure isn't very clear or documented at all.  I promise I'll spend some time on it this weekend!

Basically I have two goals:

1. Make logo size appropriate to sponsorship level.   Premier sponsors=big then descending in size going down the tiers.
2. Make logos in the same tier carry the same visual weight regardless of aspect ratio.  A very wide logo looks much smaller than a square one if they are both constrained to the same width


![Screen Shot 2019-07-05 at 6 06 55 PM](https://user-images.githubusercontent.com/281482/60747044-a2db6c80-9f50-11e9-8e00-66753ae91106.png)

![Screen Shot 2019-07-05 at 6 12 40 PM](https://user-images.githubusercontent.com/281482/60747045-a40c9980-9f50-11e9-8dc7-1d5b6fc268af.png)


Anyway this is what I'm thinking about with the logo shape values.  Basically each one represents the following approximate aspect ratios and then we can apply different dimensions to each one.  Then when adding a new logo you just pick the most appropriate shape that makes the logo look balanced with the rest in it's tier.

- square - 1:1 ish
- rectangle - 2:1 ish
- wide - 3:1 ish
- extra-wide - 4:1 ish
- super-wide - 5:1 and up

We could definitely just have a big list of rules  covering each of these shapes over all the different tiers, but I feel like there has to be a more elegant and concise way.  It's also complicated by varying view port sizes.  Need to think on it a bit more.  Let me know if you have thoughts on this too.

KyleMit commented:
### PR Stuff

@zekefarwell , so I tweaked the image and path and manipulation a little so perhaps it's a little more intuitive to us and future maintainers. 

The relative position at the beginning of the url also felt fickle `..\`.  So instead, I updated paths to be fully qualified beginning at `\src\`.  Since we're pointing to a resource location on disk, I think it makes some sense to do so explicitly, but I really am just looking for enough context to let autocomplete do its job.

I also didn't love `splice.(3)` much either.  I updated it to use a filter (`src="{{sponsor.logo | RootURL}}"`) so we can be a little more declarative with our intention.  Basically anything that has a `passthroughFileCopy` inside of src will be available at the full file path, minus `src`, because that's what we use for our input directory in `.eleventy.js`.  So the filter just removes that from any path reference and should give us a way to refer to resources using full root paths (which would be helpful if we ever changed around the url or had sub-directories that needed access to the same resources).

Hopefully that resolves any pain points enough to make this a helpful delta.

### Logo Stuff

I agree that there's some weirdness to trying to get logos with disparate aspect ratios to both:
1.  look good and 
2. have appropriate visual weight.  

Looking at [google image results for sponsors page](https://www.google.com/search?tbm=isch&q=sponsors+page) has a lot of mixed results, but underlines the point that *formatting varying aspect ratios in a grid is hard*.  Most of the aesthetically pleasing sites have uniform logo dimensions and probably solicit them explicitly from vendors to give them a particular treatment.

Perhaps there's a way to write a single set of CSS rules that don't need explicit instructions about content and can just make reasonable defaults.  But I do like your current approach of classing logos into specific aspect ratios using friendly names for a couple different breakpoints.  I think that gives us pretty reasonable control over each logo as well and sets us up to write rules for categories of things, instead of one offs.
